examples/pytorch/mnist-ddp/mnist.ipynb

```diff
@@ -0,0 +1,627 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "# PyTorch DDP Fashion MNIST Training Example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "This example demonstrates how to train a convolutional neural network to classify images using the [Fashion MNIST](https://github.com/zalandoresearch/fashion-mnist) dataset and [PyTorch DDP](https://pytorch.org/tutorials/intermediate/ddp_tutorial.html)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Install the Kubeflow SDK"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO (astefanutti): Change to the Kubeflow SDK when it's available.\n",
+    "!pip install git+https://github.com/kubeflow/training-operator.git@master#subdirectory=sdk_v2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install the PyTorch dependencies"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install torch==2.5.1\n",
+    "!pip install torchvision==0.20.0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define the training function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_fashion_mnist(params):\n",
+    "    import os\n",
+    "\n",
+    "    import torch\n",
+    "    import torch.distributed as dist\n",
+    "    import torch.nn.functional as F\n",
+    "    import torch.nn as nn\n",
+    "    import torchvision.transforms as transforms\n",
+    "    from torch.nn.parallel import DistributedDataParallel\n",
+    "    from torch.optim.lr_scheduler import StepLR\n",
+    "    from torch.utils.data import DataLoader\n",
+    "    from torch.utils.data.distributed import DistributedSampler\n",
+    "    from torchvision.datasets import FashionMNIST\n",
```

Comment:
Let's reduce number of imports as here: https://github.com/kubeflow/training-operator/blob/df640a46fa541ffdddb70e0a6a7744280c93686f/examples/pytorch/image-classification/Train-CNN-with-FashionMNIST.ipynb